tools/OpenTelemetryYoutube/OpenTelemetryYoutube/Program.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml;
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.Services;
+using Google.Apis.Util.Store;
+using Google.Apis.YouTube.v3;
+using Google.Apis.YouTube.v3.Data;
+
+namespace OpenTelemetryYoutube
+{
+    class Program
+    {
+        [STAThread]
+        static void Main(string[] args)
+        {
+            try
+            {
+                Console.WriteLine("Path to secret.json");
+                string path = Console.ReadLine();
+
+                new Program().Run(path).Wait();
+            }
+            catch (AggregateException ex)
+            {
+                foreach (var e in ex.InnerExceptions)
+                {
+                    Console.WriteLine("Error: " + e.Message);
+                }
+            }
+        }
+
+        private async Task Run(string path)
+        {
+            UserCredential credential;
+            using (var stream = new FileStream(path, FileMode.Open, FileAccess.Read))
+            {
+                credential = await GoogleWebAuthorizationBroker.AuthorizeAsync(
+                    GoogleClientSecrets.Load(stream).Secrets,
+                    new[] { YouTubeService.Scope.Youtube },
+                    "user",
+                    CancellationToken.None,
+                    new FileDataStore(this.GetType().ToString())
+                );
+            }
+
+            var youtubeService = new YouTubeService(new BaseClientService.Initializer
+            {
+                HttpClientInitializer = credential,
+                ApplicationName = this.GetType().ToString(),
+            });
+
+            var channelsListRequest = youtubeService.Channels.List("snippet,contentDetails,statistics");
+            channelsListRequest.Mine = true;
+
+            // Retrieve the contentDetails part of the channel resource for the authenticated user's channel.
+            var channelsListResponse = await channelsListRequest.ExecuteAsync();
+
+            foreach (var channel in channelsListResponse.Items)
+            {
+                await ProcessChannel(youtubeService, channel);
+            }
+        }
+
+        private static async Task ProcessChannel(YouTubeService youtubeService, Channel channel)
+        {
+            // From the API response, extract the playlist ID that identifies the list
+            // of videos uploaded to the authenticated user's channel.
```

Comment:
how does it know which channel to process?

Reply:
since i have just one channel (this one), its only processing that. i can add a constant to filter.

Reply:
thanks, this will be handy.

Reply:
done